packages/widgetbook/lib/src/routing/app_router_delegate.dart

```diff
@@ -44,8 +50,9 @@ class AppRouterDelegate extends RouterDelegate<AppRouteConfig>
         MaterialPage(
           child: _configuration.previewMode
               ? const Workbench()
-              : const WidgetbookShell(
-                  child: Workbench(),
+              : WidgetbookShell(
+                  key: key,
```

Comment:
Good job spotting where the bug is!
I suggest using a `ValueKey` of `_configuration`, instead of creating the `key` with a new `GlobalKey` each time the configuration gets updated.

```suggestion
                  key: ValueKey(_configuration),
```

Reply:
ah yes, can do that as well, I'm on it!

Reply:
changes done!